misc/toml-parser/src/main/java/io/ballerina/toml/semantic/ast/TomlTransformer.java

```diff
@@ -457,4 +491,22 @@ public TomlNode transform(BoolLiteralNode boolLiteralNode) {
     public TomlNode transform(IdentifierLiteralNode identifierLiteralNode) {
         return new TomlUnquotedKeyNode(identifierLiteralNode.value().text(), getPosition(identifierLiteralNode));
     }
+
+    private TomlNodeLocation getLocationOfKeyEntryList(List<TomlKeyEntryNode> keys) {
```

Comment:
I think we can get rid of this method. When getting the locations we can always get them from the syntax tree itself. 
Here what we do is creating locations from the AST nodes. This is prone to error. 

e.g. Suppose the order of the `TomlKeyEntryNode` nodes in the list is wrong. 
Then the new location would be messed up.

In the ST, the user's input order is always preserved whereas in AST it is not the case.

Reply:
Did as we discussed. :)